typing/typedtree.mli

```diff
@@ -250,12 +250,13 @@ and expression_desc =
       expression * Longident.t loc * Types.label_description * expression
   | Texp_array of expression list
   | Texp_ifthenelse of expression * expression * expression option
-  | Texp_sequence of expression * expression
+  | Texp_sequence of expression * Types.layout * expression
```

Comment:
This surprises me. Why is it necessary? If it's just a cached value, please add a comment saying so.

Reply:
We have layout annotations in the type tree when the layout of a subexpression isn't obvious from the expression.  This is needed to compile `void` (and maybe other layouts in some cases?) because we want to add a handler for the static exception the void thing will raise.

Here and in the loop cases, I'm allowing void things in statement positions.

Reply:
That makes sense, then. But I expect it will can be removed once we re-do the implementation of `void`. Could you maybe leave a CR-someday to revisit later?

Reply:
Indeed - I meant to say I plan to add a comment at the top explaining what the layouts are doing and what might change in the future.